### DISCUSSION TITLE
Visual Studio Designer for WPF

### DISCUSSION THREAD

**Discussion by PerfectlyFineCode** [dotnet/wpf#6827] [General]
Why is designer in Visual Studio for WPF still a thing?

The problem with that is because it uses margin as the primary tool for layouts which is all-in-all totally incorrect and beginners rely on that too much.

Designer should never be used for WPF and is a disgrace to Windows Presentation Foundation, everything about it is just terrible.
It isn't reliable, it isn't good, drag n drop shouldn't be used and will screw the developer over most of the time.

What is the reasoning for it to still exist in VS 2022?

**Comment by pchaurasia14** [comment]
While I don't know the exact reason, I suspect it may be an artifact of the days when WPF was launched as a successor to WinForms. Since most of the WinForms developers (at least the ones I knew of) relied on designer to get the layout and stuff, having a designer for WPF provided a seemingly easy transition into WPF.
